### PR TITLE
158 game selection page

### PR DESCRIPTION
**Pull request by pixelPepe**

### CODE REVIEW

File: data/gameList.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        "proposer":"0xdC5f32DEc4253Bd61092294B45AfB834C0BD2938",
+        "rules": "0xC6F81d6610A0b1bcb8cC11d50602D490b7624a96"
+    },
+    {
+        "id": 7,
+        "stake": 1,
+        "proposer": "0xdC5f32DEc4253Bd61092294B45AfB834C0BD2937",
+        "rules": "0xC6F81d6610A0b1bcb8cC11d50602D490b7624a96"
+    }
+]
```
Comment (poludnev) [inline data/gameList.json:50]: .

File: data/gameList.json
Comment (poludnev) [inline data/gameList.json:1]: What do we use this file for? Is it for test only? If yes, please put it in fixtures directory.

File: pages/games/games.module.scss [hunk 2]
```diff
@@ -1,10 +1,9 @@
 .container {
-  padding: 100px;
   text-align: center;
 
-  @media (max-width: 576px) {
-    padding: 40px 10px;
-  }
+  // @media (max-width: 576px) {
+  //   padding: 40px 10px;
+  // }
```
Comment (poludnev) [inline pages/games/games.module.scss:6]: is it supposed to be used in the future?

File: pages/index.tsx [hunk 3]
```diff
@@ -1,56 +1,280 @@
+/* eslint-disable @next/next/no-img-element */
 import type { GetStaticProps, NextPage } from 'next';
 import Head from 'next/head';
 import Image from 'next/image';
 import { useTranslation } from 'react-i18next';
 import styles from '../styles/Home.module.scss';
+// import styles from './/Home.module.scss';
 // import 'i18n/index';
```
Comment (poludnev) [inline pages/index.tsx:8]: Clean up comments if unsed.

File: pages/index.tsx [hunk 4]
```diff
@@ -1,56 +1,280 @@
...
           </div>
+          <div className={styles.mainTitle}>
+            <h2>Un-chain your game with peer-to-peer communication!</h2>
+            <p>
+              By using state channels for in-game communication, GameJutsu
+              enables multiplayer games to bypass blockchain transactions and
+              keep most of the communication off-chain, while keeping the
+              guarantees provided by smart contracts
+            </p>
+            <button onClick={() => router.push('/v2/games')}>
```
Comment (poludnev) [inline pages/index.tsx:49]: Same as above regarding handlers.

File: pages/v2/games/gameType.module.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  .gamelist {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+    margin-bottom: 10px;
+  }
+}
```
Comment (poludnev) [inline pages/v2/games/gameType.module.scss:28]: .

**Review by poludnev — CHANGES_REQUESTED** [review]

File: components/v2/GameDemo/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export * from './GameDemo'
```
[the remark this answers is not in the document]
Comment (pixelPepe) [reply components/v2/GameDemo/index.ts:1]: done

File: components/Header/Header.module.scss [hunk 7]
```diff
@@ -36,3 +36,16 @@
...
+  color: #FFFFFF;
+  background:  linear-gradient(225.67deg, #8DC8E9 3.57%, #8C5AE3 115.97%), #5F5AFA;
+  box-shadow: 0px 4px 12px rgba(0, 0, 0, 0.1);
+  border-radius: 14px;
+  margin-right: 10px;
+
+  @media (max-width: 576px) {
+    font-size: 12px;
+  }
+}
```
[the remark this answers is not in the document]
Comment (pixelPepe) [reply components/Header/Header.module.scss]: done

File: components/v2/GameThumbnail/GameThumbnail.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+            />
+            <div className={styles.address}>{shortenAddress(rulesAddress)}</div>
+          </div>
+          <div className={styles.description}>{description}</div>
+        </div>
+      </div>
+      <div className={styles.buttons}>
+        <Link
+          href={
+            address ? '/games/' + url + '?join=true' : `/connect?game=${url}`
```
[the remark this answers is not in the document]
Comment (pixelPepe) [reply components/v2/GameThumbnail/GameThumbnail.tsx]: done

File: pages/v2/games/gameType.module.scss (hunk 5 above)
Comment (pixelPepe) [reply pages/v2/games/gameType.module.scss]: done

File: components/v2/GameThumbnail/GameThumbnail.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+          }
+        >
+          <button>
+            Join <div className={styles.users}></div>
+          </button>
+        </Link>
+        <button onClick={() => setIsStartButtonOpen(!isStartButtonOpen)}>
```
[the remark this answers is not in the document]
Comment (pixelPepe) [reply components/v2/GameThumbnail/GameThumbnail.tsx]: done

File: pages/games/games.module.scss (hunk 2 above)
Comment (pixelPepe) [reply pages/games/games.module.scss:6]: these files will no longer be used. needs no fix

File: pages/index.tsx (hunk 3 above)
Comment (pixelPepe) [reply pages/index.tsx]: done

File: pages/index.tsx (hunk 4 above)
Comment (pixelPepe) [reply pages/index.tsx]: done

File: data/gameList.json
Comment (pixelPepe) [reply data/gameList.json:1]: done

File: data/gameList.json (hunk 1 above)
Comment (pixelPepe) [reply data/gameList.json]: done

File: components/v2/SelectPrize/SelectPrize.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+          No stake
+          <div className={styles.imageWrapper}>
+            <img src="/images/handshake.png" alt="handshake" />
+          </div>
+        </button>
+      </div>
+      <div className={styles.gradientBorder}>
+        <button
+          className={styles.stakeButton}
+          onClick={async (event) =>
```
[the remark this answers is not in the document]
Comment (pixelPepe) [reply components/v2/SelectPrize/SelectPrize.tsx:69]: done

File: components/v2/MyGames/MyGamesList/MyGamesList.tsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  const myGames = gameList.filter((game) => game.proposer === myAddress);
+
+  const [rulesAddress, setRulesAddress] = useState('');
+  const myFilteredGames = myGames.filter((game) => game.rules === rulesAddress);
+  useEffect(() => {
+    getRulesContract(gameType).then((response) => {
+      setRulesAddress(response.address);
+    });
+  }, []);
+  function shortenAddress(str: string) {
```
[the remark this answers is not in the document]
Comment (pixelPepe) [reply components/v2/MyGames/MyGamesList/MyGamesList.tsx]: done

File: components/v2/GameThumbnail/GameThumbnail.module.scss [hunk 12]
```diff
@@ -0,0 +1,116 @@
+.container {
+  width: 100%;
+  // height: 184px;
```
Comment (poludnev) [inline components/v2/GameThumbnail/GameThumbnail.module.scss:3]: clear unused comments

File: components/v2/SelectPrize/SelectPrize.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (pixelPepe) [reply components/v2/SelectPrize/SelectPrize.tsx:69]: done

File: gameApi/index.ts [hunk 13]
```diff
@@ -2,15 +2,15 @@ import { BigNumber, ethers } from 'ethers';
...
+import { Arbiter, GameProposedEventObject, GameStartedEventObject } from "../.generated/contracts/esm/types/polygon/Arbiter";
 import { getPolygonSdk} from "../.generated/contracts";
 import { TGameType } from 'types/game';
 import { fetchSigner } from '@wagmi/core';
 
 const provider = new ethers.providers.AlchemyProvider('matic', process.env.NEXT_PUBLIC_ALCHEMY_API_KEY!);
 const getSdk = async () => getPolygonSdk(await getSigner())
 const getSdkRead = () => getPolygonSdk(provider);
-export const getArbiter = async () => (await getSdk()).arbiter;
+export const getArbiter: ()=>Promise<Arbiter> = async () => (await getSdk()).arbiter;
```
Comment (poludnev) [inline gameApi/index.ts:13]: it is simpler to set only the returning type, not to describe the full type of a function: 
getArbiter = async (): Promise<Arbiter> => {...}

File: pages/_app.tsx [hunk 14]
```diff
@@ -14,11 +15,16 @@ import '@rainbow-me/rainbowkit/styles.css';
...
+
+// const { chains, provider, webSocketProvider } = configureChains(
+//   [polygon],
+//     [alchemyProvider({ apiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY! })],
+// );
 
 const { chains, provider, webSocketProvider } = configureChains(
   [polygon],
-  [alchemyProvider({ apiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY! })]
+  [publicProvider()]
```
Comment (poludnev) [inline pages/_app.tsx:27]: Don't forget to return alchemy for prod.

File: pages/games/games.module.scss (hunk 2 above)
Comment (poludnev) [reply pages/games/games.module.scss:6]: If not in use any more, clear them out

**Review by poludnev — APPROVED** [review]

File: pages/_app.tsx (hunk 14 above)
Comment (pixelPepe) [reply pages/_app.tsx]: done

File: gameApi/index.ts (hunk 13 above)
Comment (pixelPepe) [reply gameApi/index.ts:13]: done

**Pull request merged**